.github/workflows/ci.yml

```diff
@@ -33,6 +33,60 @@ jobs:
     - name: Run pre-commit
       run: |
         pre-commit run --verbose --all-files
-    - name: Run unit tests and generate coverage report
+    - name: Run unit tests
       run: |
-        python -m pytest --cov --cov-config=.coveragerc ./syntheseus/tests/
+        coverage run -p -m pytest ./syntheseus/tests/
+        coverage report --data-file .coverage.*
+    - name: Upload coverage report
+      uses: actions/upload-artifact@v4
+      with:
+        name: .coverage.core-${{ matrix.python-version }}
+        path: .coverage.*
+  test-models:
```

Comment:
Are we running the single-step model tests separately to avoid re-running with multiple python versions (to save compute), or because they only work with a very specific python version?

Reply:
The single step models tests use full a syntheseus installation, whereas the core tests use the minimal installation. By separating the two we get to test both setups. But yes, it's also true that the single-step models have pinned dependencies and we would need to define separate pins to run them under Python 3.7 (although they may actually run, I haven't checked).

Reply:
Makes sense. Let's just keep it like this then. Creating a full python 3.7 environment with all dependencies does not sound like a useful exercise to me (especially since 3.7 is deprecated now).